### PR TITLE
feat(oas): check examples are valid against their schema

### PR DESCRIPTION
**Pull request by philsturgeon**
If an example is provided, make sure it is valid against the schema for that property.

SL-2065

### Please check if the PR fulfills these requirements
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?

Feature

### What is the current behavior?

Example is pretty much ignored.

### If this is a feature change, what is the new behavior?

Now examples with nonsense values, formats that are not matched, etc, will product validation errors. 

### Does this PR introduce a breaking change?

This will create a breaking change if you have invalid examples, so you will want to fix the examples.

### CODE REVIEW

File: src/functions/schema-path.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  if (opts.field && typeof targetVal === 'object') {
+    value = targetVal[opts.field];
+  }
+
+  if (!value) return [];
+
+  // Hunt in the original targetVal for the schema we are going to apply
+  let schemaObj;
+  try {
+    schemaObj = jp.value(targetVal, opts.schemaPath);
```
Comment (casserni) [inline src/functions/schema-path.ts]: Should we remove the field value we are testing on out of the `schemaObj`?

So if we are testing the `example` property in a schema should we remove that property from the schema

Comment (casserni) [reply src/functions/schema-path.ts]: `delete schemaObj[opts.field]`

File: src/types/rule.ts [hunk 2]
```diff
@@ -82,3 +82,10 @@ export interface ISchemaOptions {
   schema: object;
 }
 export type SchemaRule = IRule<RuleFunction.SCHEMA, ISchemaOptions>;
+
+export interface ISchemaPathOptions {
+  schemaPath: string;
+  // the `path.to.prop` to field, or special `@key` value to target keys for matched `given` object
+  field?: string;
```
Comment (casserni) [inline src/types/rule.ts]: Can field also be a path? So we aren't limited to top level properties?

File: src/functions/schema-path.ts (hunk 1 above)
Comment (philsturgeon) [reply src/functions/schema-path.ts]: It doesnt hurt having it there. It's being ignored by the validator, which is why we need this rule. :)

File: src/types/rule.ts (hunk 2 above)
Comment (casserni) [reply src/types/rule.ts]: TODO/feature request so resolving

**Pull request merged**